### PR TITLE
fix: trace sampling rules order

### PR DESCRIPTION
**Pull request by dmehala**
# Description

Reset trace sampling rules legacy order.

### CODE REVIEW

File: src/datadog/config_manager.cpp [hunk 1]
```diff
@@ -98,7 +108,19 @@ std::vector<ConfigMetadata> ConfigManager::update(const ConfigUpdate& conf) {
 
   std::lock_guard<std::mutex> lock(mutex_);
 
-  decltype(rules_) rules;
+  // NOTE(@dmehala): Sampling rules are generally not well specified.
+  //
+  // Rules are eveluated in the order they are inserted, which means the most
```
Comment (dmehala) [inline src/datadog/config_manager.cpp:113]:
```suggestion
  // Rules are evaluated in the order they are inserted, which means the most
```

File: src/datadog/config_manager.cpp [hunk 2]
```diff
@@ -42,7 +43,10 @@ Expected<Rules> parse_trace_sampling_rules(const nlohmann::json& json_rules) {
         return *error;
       }
 
-      rate.value = *maybe_rate;
+      rule.rate = *maybe_rate;
+    } else {
+      return Error{Error::TRACE_SAMPLING_RULES_INVALID_JSON,
+                   "Missing \"sample_rate\" field"};
```
Comment (pablomartinezbernardo) [inline src/datadog/config_manager.cpp:49]: qq for this and other new error handling: Is a default value set somewhere? Or will parsing now fail when it didn't before?

File: src/datadog/config_manager.cpp [hunk 3]
```diff
@@ -98,7 +108,19 @@ std::vector<ConfigMetadata> ConfigManager::update(const ConfigUpdate& conf) {
 
   std::lock_guard<std::mutex> lock(mutex_);
 
-  decltype(rules_) rules;
+  // NOTE(@dmehala): Sampling rules are generally not well specified.
+  //
+  // Rules are eveluated in the order they are inserted, which means the most
+  // specific matching rule moght not be evaluated, even though it should be.
```
Comment (pablomartinezbernardo) [inline src/datadog/config_manager.cpp:114]:
```suggestion
  // specific matching rule might not be evaluated, even though it should be.
```

**Review by pablomartinezbernardo — APPROVED** [review]
LGTM

File: src/datadog/config_manager.cpp (hunk 2 above)
Comment (dmehala) [reply src/datadog/config_manager.cpp:49]: Parsing will fail, those are mandatory fields.

**Pull request merged**